app/services/decidim/action_delegator/sms_gateway.rb

```diff
@@ -5,23 +5,44 @@ module ActionDelegator
     class SmsGateway
       attr_reader :mobile_phone_number, :code, :response
 
+      SMS_GATEWAY_PROVIDER_JOBS = {
+        som_connexio: SendSmsJob,
+        twilio: TwilioSendSmsJob
+      }.freeze
+
       def initialize(mobile_phone_number, code)
         @mobile_phone_number = mobile_phone_number
         @code = code
       end
 
       def deliver_code
-        SendSmsJob.perform_later(sender_name, mobile_phone_number, message)
+        return false unless sms_gateway_provider_valid?
+
+        sms_gateway_job.perform_later(sender_name, mobile_phone_number, message)
 
         true
       end
 
       private
 
+      def sms_gateway_job
+        SMS_GATEWAY_PROVIDER_JOBS[sms_gateway_provider.to_sym]
+      end
+
       def sender_name
         ENV["SMS_SENDER_NAME"] || Decidim.application_name
```

Comment:
I'd also rename this to `SMS_SENDER` so I don't get confused with its meaning when configuring it for Twilio.

Reply:
Also, IMO it's better we don't provide a default so we purposefully configure it properly depending on the provider.